src/test/java/org/json/junit/JSONObjectNumberTest.java

```diff
@@ -23,7 +23,10 @@ public class JSONObjectNumberTest {
     @Parameters(name = "{index}: {0}")
     public static Collection<Object[]> data() {
         return Arrays.asList(new Object[][]{
-            {"{value:50}", 1},
+            {"{value:0050}", 1},
+            {"{value:0050.0000}", 1},
+            {"{value:-0050}", -1},
+            {"{value:-0050.0000}", -1},
```

Comment:
Add a test cases for:
* 0.5 -> expects "0.5"
* -0.5 -> expects "-0.5"
* 00.05 -> expects "0.05"
* -00.05 -> expects "-0.05"

Reply:
also add:

* `.5` -> expects ".5"
* `-.5` -> expects "-.5"

Reply:
Also add test case for:
* `00000` -> Expects "0"

Reply:
For decimal cases, the numbers which does not have anything before decimal point, e.g., .5, -.5 etc, what should be the behavior when getLong is called.
As of now, they throw  NumberFormatException. It works for optLong though.
@johnjaylward  @stleary

Reply:
I believe in the Issue it was noted that the `opt` parsing was preferred, and I tend to agree.